src/feature/control/control_fmt.c

```diff
@@ -165,6 +171,74 @@ circuit_describe_status_for_controller(origin_circuit_t *circ)
   return rv;
 }
 
+/**
+ * Extends the description of a <b>circ</b>'s status.
+ */
+void
+circuit_extend_status_for_controller(const origin_circuit_t *circ,
+                                     smartlist_t *descparts)
+{
+  // Supplementary data
+  smartlist_add_asprintf(descparts, "UNUSABLE_FOR_NEW_CONNS=%u",
+                         circ->unusable_for_new_conns);
+  smartlist_add_asprintf(descparts, "IDLE_TIMEOUT=%d",
+                         circ->circuit_idle_timeout);
+  smartlist_add_asprintf(descparts, "EARLY_CELLS_SENT=%u",
+                         circ->relay_early_cells_sent);
+  smartlist_add_asprintf(descparts, "REMAINING_RELAY_EARLY_CELLS=%u",
+                         circ->remaining_relay_early_cells);
+  smartlist_add_asprintf(descparts, "PADDING_NEGOTIATION_FAILED=%u",
+                         circ->padding_negotiation_failed);
+  smartlist_add_asprintf(descparts, "PATH_STATE=%s",
+                         pathbias_state_to_controller_string(circ->path_state)
+                         );
+  smartlist_add_asprintf(descparts, "HAS_RELAXED_TIMEOUT=%u",
+                         circ->relaxed_timeout);
+  smartlist_add_asprintf(descparts, "IS_ANCIENT=%u", circ->is_ancient);
+  smartlist_add_asprintf(descparts, "HAS_OPENED=%u", circ->has_opened);
+  smartlist_add_asprintf(descparts, "HS_HAS_TIMED_OUT=%u",
+                         circ->hs_circ_has_timed_out);
+
+  // Prepend policy data
+  const char *prepend_policy_joined;
```

Comment:
`NULL` initialize?

Reply:
The variable is now initialized every time, thanks for the comment.